### ISSUE TITLE
Update Component Registration table with componentVersion and outputVersion

### ISSUE THREAD

**Issue by jrobble** [openmpf/openmpf#1702]
We should get into the habit of updating the `componentVersion` for each component whenever there is a change. I see no reason why that necessarily has to match the Docker tag, other than to avoid user confusion.

If showing different versions on the Component Registration page in the web UI will be too confusing, I think we can use the `middlewareVersion` field to populate the `Version` column in the table on that page. Basically, the whole point of that field is to represent compatibility, which is the only thing I imagine that the user does currently when looking at the table on that page. We can always keep the `middlewareVersion` and Docker tag in sync.

Better yet, we can update the table so that instead of a just a `Version` column, we instead have a `Compatibility Version` column and `Component Version` column to represent the `middlewareVersion` and `componentVersion` respectively. Maybe even have a column for `outputVersion`.

---

Note that it's important to keep `componentVersion` and `outputVersion` separate. Consider these scenarios:

1. We implement a hotfix or feature that **does not** require previous jobs to be rerun. We update the `componentVersion` according to the semantic versioning rules and don't change the `outputVersion`.

2. We implement a hotfix or feature that **does** require previous jobs to be rerun. We update the `componentVersion` according to the semantic versioning rules and update the `outputVersion`.

Basically, `outputVersion` is just a monotonically increasing dirty flag.
